tokio-current-thread/src/lib.rs

```diff
@@ -64,6 +65,9 @@ pub struct CurrentThread<P: Park = ParkThread> {
 
     /// Receiver for futures spawned from other threads
     spawn_receiver: mpsc::Receiver<Box<Future<Item = (), Error = ()> + Send + 'static>>,
+
+    /// The thread-local ID assigned to this executor.
+    id: usize,
```

Comment:
I wonder if using `u64` would be better ‒ that one would take very long to overflow, but usize is 32bits on some systems and that conceivably could (and likely do it in production with release, not while testing). While unlikely someone would rotate 4 billion executors on a single thread, being on the safe side looks cheap here.